include/Aulib/Decoder.h

```diff
@@ -25,6 +25,7 @@ class AULIB_EXPORT Decoder
     Decoder(const Decoder&) = delete;
     auto operator=(const Decoder&) -> Decoder& = delete;
 
+#if defined(__cplusplus) && __cplusplus >= 201603
```

Comment:
This can simply be `#if __cplusplus >= 201603`. No need to check if it's defined. It's always defined since this is a C++-only header.